### ISSUE TITLE
Highlighting misspelled commands

### ISSUE THREAD

**Issue by gergelyk** [nushell/nushell#11128]
### Related problem

While typing command in the shell, I'm unable to say if I spelled it correctly until I hit enter. In case of a mistake, I need to fetch the last item from the history and fix the typo.

### Describe the solution you'd like

It would be great if nushell could highlight misspelled commands while typing, so that we know about typos before even hitting enter.

### Additional context and details

Idea taken from [elvish](elv.sh).

**Comment by fdncred** [comment]
it does this already but it's subtle. correctly spelled internal commands are colored with the color of `shape_internal`. so you can change that color with any color in `ansi --list` and better see when you've found a correctly spelled command. for externals, i don't think nushell is aware of them enough to color them precisely. I think anything you type in the command position is considered "shape_external" until it's resolved.

**Comment by gergelyk** [comment]
Thanks @fdncred. Perhaps you mean `shape_internalcall`. I confirm, it works for internal command. Do you think nushell could use `which` to determine if an external command of given name is available? Or would it be too much of an impact on performance? Of course, args of the external commands couldn't be validated as they are in case of internal commands.

**Comment by fdncred** [comment]
Right, `shape_internalcall` my mistake. I have the same concern that which would be too impactful but the only way to tell is to try it. I was thinking we could have a `shape_internal_resolved` to trigger a different color.

**Comment by evelant** [comment]
It looks like the new `shape_external_resolved` isn't working, at least on macos with nu 0.88.1. I have the following in my config but no externals get resolved, they're always orange even when valid
```
$env.config.highlight_resolved_externals = true
      $env.config.color_config.shape_internalcall = cyan_bold               # internal commands that are found will be this color
      $env.config.color_config.shape_external = darkorange                  # external "commands" that do not exist, will be this color
      $env.config.color_config.shape_external_resolved = light_yellow_bold  # external commands that are found with `which`, will be this color
```

**Comment by fdncred** [comment]
@evelant it's still working for me on Windows, WSL Ubuntu, and MacOS with 0.88.2

**Comment by weirdan** [comment]
This is possible now, for both builtin (or custom) and external commands. I think the feature request should be considered implemented and any bugs (should there be some) should be reported separately.
